testsuite/MDAnalysisTests/core/test_atomselections.py

```diff
@@ -602,12 +588,38 @@ def test_cyzone(self, u, meth, periodic):
 
         assert ref == set(result.indices)
 
+    @pytest.mark.parametrize('input_val, expected',
+                             zip(BaseDistanceSelection.methods,
+                                 [25, 31, 33, 25]))
```

Comment:
So this means that kdtree and non-kdtree are returning a different result for periodic=True

Reply:
Does that mean more changes are needed or we need to think about this a bit more?